src/java.desktop/share/classes/javax/swing/ImageIcon.java

```diff
@@ -328,7 +328,6 @@ protected void loadImage(Image image) {
             try {
                 mTracker.waitForID(id, 0);
             } catch (InterruptedException e) {
-                System.out.println("INTERRUPTED while loading Image");
```

Comment:
should be ok. Only question is, should we set the Thread interrupt flag or MediaTracker ABORTED flag too here but seems like this file does not check for error status after calling loadImage and it's been there for long so it might affect existing application using ImageIcon.

Reply:
That was my concern too. In fact, `InterruptedException` is not in the regular flow, it indicates a somewhat error condition which isn't actually handled.

If `loadStatus` of the image is not `COMPLETE` or `ERRORED`, we should probably move it to `ABORTED`.

I don't think we should raise Thread.interrupt flag again.

Reply:
WaitForID() definition states that "If there is an error while loading or scaling an image, then that image is considered to have finished loading". Does this indicates that no error checking is required.....?

Reply:
But in this case it's the waiting thread that is interrupted. The `MediaTracker` could complete loading the image successfully yet `ImageIcon` instance wouldn't update its `loadStatus` because it removes itself from the media tracker.

Reply:
We shouldn't be setting any MediaTracker status here - this is the caller ! It would be absolutely wrong. And it seems likely to me that this println was put there by the developer of this code who was him/herself not sure if this would occur and what to do about it so wanted to have a message if it ever occurred so they could figure out how to handle it.
It seems to me that if the thread were interrupted there'd only be a very narrow window in which it likely that the image were complete and that if interrupted then I agree with what I think Alexey is saying which is that the MediaTracker may not have had a chance to even update status since it was interrupted .. depends how sophisticated the implementation is.
Since ImageIcon.getLoadStatus() specifies only 3 possible return values then yes, we should ensure it is one of those . So if it is LOADING then set it to ABORTED .. However a possible wrinkle is that this call being interrupted may not actually prevent the separate threads the MediaTracker kicks off from continuing .. so it may actually be still loading. And it is just that the waiting was interrupted.
Someone would need to dig into MediaTracker to check.
But since the code removes the image from the tracker .. maybe we don't care ?

Reply:
Since the image is removed from MediaTracker, its status can't change after that. Therefore `LOADING` status should be changed to `ABORTED`.
[JDK-6421373](https://bugs.openjdk.java.net/browse/JDK-6421373) has a test case which can be modified to load an image and check `loadStatus` of the image.

Reply:
Ok, will modify the code and check the status.

Reply:
@aivanov-jdk , even after interrupted the load status gets the "COMPLETE" status and I'm able to use the Imageicon for frame......

Reply:
@TejeshR13, What kind of image do you load?

If I load an image from the Internet, I get LOADING status. I've attached `ImageIconInterrupt.java` to [JDK-6421373](https://bugs.openjdk.java.net/browse/JDK-6421373). It produces the following output:

```
INTERRUPTED while loading Image
false
1
    LOADING
```

Reply:
I have loaded gif image from local storage.....
```
INTERRUPTED while loading Image
false
    COMPLETE
```

Reply:
A small file from local storage loads fast enough. By the time you get to read status, load could complete.

For the url, do you use the same URL? If the connection can be established, it should load pretty fast too. However, if you need a proxy server to access the Internet but you won't provide one, it won't load as fast, nor will it report error fast enough.

Anyway, my example above confirms *it's possible the status remains `LOADING`*. But the image isn't loading, it will never load and its status will never change, so you have to amend the status to a correct one.

Reply:
Yeah, got it...... I used the same url though......

Reply:
N for current bug, where to change to change the status to "ABORTED"......?? As I cannot change in ImageIcon class......

Reply:
@aivanov-jdk , U r right, the status remains LOADING for url...... I was getting "ERRORED" because I was showing a message using `JOptionPane.showMessageDialog(null, "");` before setting ImageIcon.

Reply:
> @aivanov-jdk , U r right, the status remains LOADING for url...... I was getting "ERRORED" because I was showing a message using `JOptionPane.showMessageDialog(null, "");` before setting ImageIcon.

Ah, with the message dialog shown there's enough time to complete loading the image.

Reply:
> N for current bug, where to change to change the status to "ABORTED"......?? As I cannot change in ImageIcon class......

Why can't you change the status in `ImageIcon`? It's where you change it.

```java
            loadStatus = mTracker.statusID(id, false);
            mTracker.removeImage(image, id);
            if (loadStatus & MediaTracker.LOADING != 0) {
                loadStatus = MediaTracker.ABORTED;
            }
```
The first two lines already exist, right after the catching `InterruptedException`. You just need to add the `if` block to amend the status.

Reply:
How about setting the status to ABORTED only if it is Interrupted.....?
`            try {
                mTracker.waitForID(id, 0);
            } catch (InterruptedException e) {
                bIsInterrupted = true;
            }
            if (bIsInterrupted == true)
            {
                loadStatus = MediaTracker.ABORTED;
            }
            else
            {
                loadStatus = mTracker.statusID(id, false);
            }`

Reply:
That's a good point. However, you have get the status from `MediaTracker` first: the image could finish loading before the thread was interrupted.

```java
        try {
            mTracker.waitForID(id, 0);
        } catch (InterruptedException e) {
            interrupted = true;
        }

        loadStatus = mTracker.statusID(id, false);
        mTracker.removeImage(image, id);
        if (interrupted && (loadStatus & MediaTracker.LOADING != 0)) {
            loadStatus = MediaTracker.ABORTED;
        }
```

On the other hand, if the thread isn't interrupted, `waitForID` returns only after the image moves into its final state: `LOADED`, `ABORTED`, or `ERRORED`; if its status is still `LOADING`, the thread is interrupted.

Therefore my original suggestion is still correct: the status is amended only when the thread is interrupted. Yet the intention of the code seems clearer with the additional `interrupted` flag. At the same time, I'm still inclined towards the shorter version without the flag.